idaes/core/initialization/initializer_base.py

```diff
@@ -0,0 +1,399 @@
+    def get_logger(self, model):
+        return idaeslog.getInitLogger(model.name, self.config.output_level)
+
+    def initialize(
+        self, model: Block, initial_guesses: dict = None, json_file: str = None
+    ):
+        """
+        Execute full initialization routine.
+
+        Args:
+            model: Pyomo model to be initialized.
+            initial_guesses: dict of initial guesses to load.
+            json_file: file name of json file to load initial guesses from as str.
+
+        Note - can only provide one of initial_guesses or json_file.
+
+        Returns:
+            InitializationStatus Enum
+        """
+        # 1. Get current model state
+        self.get_current_state(model)
+
+        # 2. Load initial guesses
+        self.load_initial_guesses(
+            model, initial_guesses=initial_guesses, json_file=json_file
+        )
+
+        # 3. Fix states to make square
+        self.fix_initialization_states(model)
+
+        # 4. Prechecks
+        self.precheck(model)
+
+        # 5. try: Call specified initialization routine
+        try:
+            results = self.initialization_routine(model)
+        # 6. finally: Restore model state
+        finally:
+            self.restore_model_state(model)
+
+        # 7. Check convergence
+        return self.postcheck(model, results_obj=results)
+
+    def get_current_state(self, model: Block):
+        """
+        Get and store current state of variables (fixed/unfixed) and constraints/objectives
+        activated/deactivated in model.
+
+        Args:
+            model: Pyomo model to get state from.
+
+        Returns:
+            dict serializing current model state.
+        """
+        self.initial_state = to_json(model, wts=StoreState, return_dict=True)
+
+        return self.initial_state
+
+    def load_initial_guesses(
+        self, model: Block, initial_guesses: dict = None, json_file: str = None
+    ):
+        """
+        Load initial guesses for variables into model.
+
+        Args:
+            model: Pyomo model to be initialized.
+            initial_guesses: dict of initial guesses to load.
+            json_file: file name of json file to load initial guesses from as str.
+
+        Note - can only provide one of initial_guesses or json_file.
+
+        Returns:
+            None
+
+        Raises:
+            ValueError if both initial_guesses and json_file are provided.
+        """
+        if initial_guesses is not None and json_file is not None:
+            self.status = InitializationStatus.Error
+            raise ValueError(
+                "Cannot provide both a set of initial guesses and a json file to load."
+            )
+
+        if initial_guesses is not None:
+            self._load_values_from_dict(model, initial_guesses)
+        elif json_file is not None:
+            # TODO: What should we load here?
+            # As this is just initialization, for now I am only loading variable values
+            from_json(
```

Comment:
I think you are right.  Non-fixed variable values and nothing else.  There are good reasons to load other stuff, but I think that would be something other than initialization as we define it for IDAES.

Reply:
That was my thought - for the purposes of initialization I felt we should keep this limited. Users can just call `from_json` directly if they want to load more.